File: javascript/util/palindrome.js
const digits = require(`./digits`);

function isArrayPalindrome(arr) {
  let i = 0;
  let j = arr.length - 1;
  while (i < j) {
    if (arr[i] !== arr[j]) {
      return false;
    }
    i++;
    j--;
  }

  return true;
}

function isPalindrome(num) {
  return isArrayPalindrome(digits.getDigitsReversed(num));
}

module.exports = {
  isArrayPalindrome,
  isPalindrome,
};

File: python/util/palindrome.py
from . import digits


def is_array_palindrome(arr):
  i = 0
  j = len(arr) - 1
  while i < j:
    if arr[i] != arr[j]:
      return False
    i += 1
    j -= 1

  return True


def is_palindrome(num):
  return is_array_palindrome(digits.get_digits_reversed(num))
